File: sdk/samples/all in on code/Visual Studio 2008/CSWPFDataBinding/Converters.cs
/************************************* Module Header **************************************\
* Module Name:  Converters.cs
* Project:      CSWPFDataBinding
* Copyright (c) Microsoft Corporation.
* 
* This example demonstrates how to use DataBinding in WPF
* 
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* History:
* * 10/29/2009 3:00 PM Bruce Zhou Created
 * 
\******************************************************************************************/
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Data;

namespace CSWPFDataBinding
{
    public class SalaryFormmatingConverter:IValueConverter
    {

        #region IValueConverter Members

        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            double dolloars =0;
            string formattedSalary = string.Empty;
            
            if (value == null)
            {
                throw new NullReferenceException("value can not be null");
            }
            else
            {
                Double.TryParse(value.ToString(), out dolloars);
            }
            formattedSalary = String.Format("Total={0}$", dolloars);
            return formattedSalary;
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
             double dolloars =0;
            if (value == null)
            {
                throw new NullReferenceException("value can not be null");
            }

            Double.TryParse(value.ToString().TrimStart(new Char[] {'T', 't', 'o', 'a', 'l', '='}).TrimEnd('$'), out dolloars);        
            return dolloars;
        }

        #endregion
    }
}

File: sdk/samples/all in on code/Visual Studio 2008/VBWPFDataBinding/Converters.vb
'****************************** Module Header ******************************'
' Module Name:  Converters.vb
' Project:      VBWPFDataBinding
' Copyright (c) Microsoft Corporation.
' 
' This example demonstrates how to use DataBinding in WPF
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************'

Public Class SalaryFormmatingConverter
    Implements IValueConverter

#Region "IValueConverter Members"

    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Dim dolloars As Double = 0
        Dim formattedSalary As String = String.Empty

        If value Is Nothing Then
            Throw New NullReferenceException("value can not be null")
        Else
            [Double].TryParse(value.ToString(), dolloars)
        End If
        formattedSalary = [String].Format("Total={0}$", dolloars)
        Return formattedSalary
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Dim dolloars As Double = 0
        If value Is Nothing Then
            Throw New NullReferenceException("value can not be null")
        End If
        [Double].TryParse(value.ToString().TrimStart(New Char() {"T"c, "t"c, "o"c, "a"c, "l"c, "="c}).TrimEnd("$"c), dolloars)
        Return dolloars
    End Function

#End Region
End Class